primitives/core/src/traits.rs

```diff
@@ -40,9 +59,6 @@ pub trait BareCryptoStore: Send + Sync {
 		id: KeyTypeId,
 		seed: Option<&str>,
 	) -> Result<sr25519::Public, String>;
-	/// Returns the sr25519 key pair for the given key type and public key combination.
-	fn sr25519_key_pair(&self, id: KeyTypeId, pub_key: &sr25519::Public) -> Option<sr25519::Pair>;
-
 	/// Returns all ed25519 public keys for the given key type.
 	fn ed25519_public_keys(&self, id: KeyTypeId) -> Vec<ed25519::Public>;
```

Comment:
Should we remove these methods now? I find it weird that we now have two different set of APIs, returning different types but essentially doing the same thing. Happy to have it done in a follow up PR, don't want to block this one.

Reply:
This is kinda related to my comment here:
https://github.com/paritytech/substrate/pull/4925/files#r384537219

What we can do is generalize them a bit by renaming those methods to something like `public_keys(CryptoTypeId, KeyTypeId)` but we're going to be left by functions such as `sr25519_generate_new` and similar functions which also need to somehow be generalized in the same fashion. This gets the scope of the PR a bit bigger though.

Reply:
Fine to do a clean-up in a second PR, but please create an issue for this after the PR is merged.